pkg/pex/pex.go

```diff
@@ -358,6 +347,71 @@ func limit(opts *discovery.Options, is []peer.AddrInfo) []peer.AddrInfo {
 	return is
 }
 
+func (px *PeerExchange) namespace(nss string) namespace {
+	px.mu.Lock()
+	defer px.mu.Unlock()
+
+	ns, ok := px.ns[nss]
+
+	if !ok {
+		ns = px.createNamespace(nss)
+		px.ns[nss] = ns
+	}
+
+	return ns
+}
+
+func (px *PeerExchange) trackNamespace(nss string, ttl time.Duration) {
+	for {
+		select {
+		case px.namespace(nss).ttl <- ttl:
+			return
+		case <-px.namespace(nss).ctx.Done():
```

Comment:
Shouldn't this return if the namespace's context expires?

Reply:
It shouldn't, cause if the namespace context is Done, it should continue the loop, until a new namespace is created and the desired TTL is set. 

Check the latest push, the loop has been refactored and now it makes it clearer how it works